src/Arriba/Arriba.Composition/Composition/Host.cs

```diff
@@ -3,93 +3,76 @@
 
 using System;
 using System.Collections.Generic;
-using System.Composition.Convention;
-using System.Composition.Hosting;
 using System.Linq;
+using Arriba.Caching;
 using Arriba.Communication;
+using Arriba.Communication.Application;
 using Arriba.Communication.ContentTypes;
+using Arriba.Communication.Server.Application;
+using Arriba.Configuration;
 using Arriba.Model;
+using Arriba.Model.Correctors;
+using Arriba.Server;
+using Arriba.Server.Application;
+using Arriba.Server.Authentication;
+using Arriba.Server.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
 
 namespace Arriba.Composition
 {
-    public class Host : IDisposable
+    public static class HostExtensions
     {
-        private CompositionHost _container = null;
-        private ContainerConfiguration _configuration = null;
-
-        public Host()
+        public static void AddArribaServices(this IServiceCollection services, ISecurityConfiguration config)
         {
-            ArribaServices.Initialize();
-            var conventions = new ConventionBuilder();
-            conventions.ForTypesDerivedFrom<IChannel>()
-                .ExportInterfaces()
-                .Shared();
-
-            conventions.ForTypesDerivedFrom<IContentReader>()
-                .Export<IContentReader>()
-                .Shared();
+            services.AddSingleton<ISecurityConfiguration>(config);
+         
+            var arribaTypes = GetArribaTypes();
+            services.AddDerivedTypes<IChannel>(arribaTypes);
+            services.AddDerivedTypes<IContentReader>(arribaTypes);
+            services.AddDerivedTypes<IContentWriter>(arribaTypes);
+            services.AddDerivedTypes<JsonConverter>(arribaTypes);
```

Comment:
Not loving this.  It looks to be doing what the old composition framework used to, but I don't see a reason to keep the type scanning in.  Can we add them as transients explicitly?  I don't think there are that many and then it would like the plain old .net di container.  A temporary trace in AddDerivedTypes would tell us exactly what to add

Reply:
Good point. I followed this patter to keep the code as close to the existing version as possible, however I agree that it's not a good practice and will update.